tests/everflow/everflow_test_utilities.py

```diff
@@ -60,74 +74,45 @@ def setup_info(duthosts, rand_one_dut_hostname, tbinfo):
     switch_capability_facts = duthost.switch_capabilities_facts()["ansible_facts"]
     acl_capability_facts = duthost.acl_capabilities_facts()["ansible_facts"]
 
+    topo_type = tbinfo["topo"]["type"]
     # Get the list of T0/T2 ports
     for dut_port, neigh in mg_facts["minigraph_neighbors"].items():
-        if "t1" in topo:
-            # Get the list of T0/T2 ports
-            if "t0" in neigh["name"].lower():
-                # Add Tor ports to namespace
-                tor_ports_namespace_map[neigh['namespace']].append(dut_port)
-                tor_ports_namespace.add(neigh['namespace'])
-            elif "t2" in neigh["name"].lower():
-                # Add Spine ports to namespace
-                spine_ports_namespace_map[neigh['namespace']].append(dut_port)
-                spine_ports_namespace.add(neigh['namespace'])
-        elif "t0" in topo:
-            # Get the list of Server/T1 ports
-            if "server" in neigh["name"].lower():
-                # Add Server ports to namespace
-                server_ports_namespace_map[neigh['namespace']].append(dut_port)
-                server_ports_namespace.add(neigh['namespace'])
-            elif "t1" in neigh["name"].lower():
-                # Add T1 ports to namespace
-                t1_ports_namespace_map[neigh['namespace']].append(dut_port)
-                t1_ports_namespace.add(neigh['namespace'])
-        else:
-            # Todo: Support dualtor testbed
-            pytest.skip("Unsupported topo")
-
-    if 't1' in topo:
-        # Set of TOR ports only Namespace 
-        tor_only_namespace = tor_ports_namespace.difference(spine_ports_namespace)
-        # Set of Spine ports only Namespace 
-        spine_only_namespace = spine_ports_namespace.difference(tor_ports_namespace)
-        # Randomly choose from TOR_only Namespace if present else just use first one 
-        tor_namespace = random.choice(tuple(tor_only_namespace)) if tor_only_namespace else tuple(tor_ports_namespace)[0]
-        # Randomly choose from Spine_only Namespace if present else just use first one 
-        spine_namespace = random.choice(tuple(spine_only_namespace)) if spine_only_namespace else tuple(spine_ports_namespace)[0]
-        tor_ports = tor_ports_namespace_map[tor_namespace]
-        spine_ports = spine_ports_namespace_map[spine_namespace]
-
+        pytest_assert(topo_type in UPSTREAM_NEIGHBOR_MAP and topo_type in DOWNSTREAM_NEIGHBOR_MAP, "Unsupported topo")
+        if UPSTREAM_NEIGHBOR_MAP[topo_type] in neigh["name"].lower():
+            upstream_ports_namespace_map[neigh['namespace']].append(dut_port)
+            upstream_ports_namespace.add(neigh['namespace'])
+        elif DOWNSTREAM_NEIGHBOR_MAP[topo_type] in neigh["name"].lower():
+            downstream_ports_namespace_map[neigh['namespace']].append(dut_port)
+            downstream_ports_namespace.add(neigh['namespace'])
+
+    if 't1' in topo or 'm0' in topo:
+        # Set of downstream ports only Namespace
```

Comment:
For M0 device, do we need to consider the namespace? We need to do that for T1 device because the T2 can be a multi ASIC device. But for M0's upsteam, do we have multi ASIC?

Reply:
Thanks for pointing out that! Fixed.